### DISCUSSION TITLE
launch.json do not work during python debug

### DISCUSSION THREAD

**Discussion by HumanZhong** [coder/code-server#4842] [Q&A]
1. create a `launch.json` file
```
{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": [
        {
            "name": "multigpu_debug",
            "type": "python",
            "request": "launch",
            "program": "${file}",
            "console": "integratedTerminal",
            "env": {
                "OMP_NUM_THREADS": "1"
            },
            "args": [
                "--config-file",
                "debug.yaml",
                "--num-gpus",
                "4"
            ]
        },
    ]
}
```
2. set some predefined parameters like 'args' and 'env'
3. click the little green button to 'run and debug'
![image](https://user-images.githubusercontent.com/31882926/153226829-327897cc-a2e0-41e7-82fe-9b8276b62b60.png)
4. comes out a window saying 'Select a debug configuration' (as I am concerned, there should be no need to select this since I have already created a `launch.json`)
5. select 'Python File (Debug the currently activate Python file)'
![image](https://user-images.githubusercontent.com/31882926/153226349-efd1476e-0eae-4003-8be7-6a3b20b2a271.png)
6. debug starts, **_but there are no parameters sent into the program_**
7. find the user setting of `launch` and move the configurations in `launch.json` to user setting 
![image](https://user-images.githubusercontent.com/31882926/153227310-5736a13d-c4d3-48fc-b716-e4b0e31b7d0c.png)
8. debug works fine and the parameters are properly sent to the program.

Question: What is the possible reason for `launch.json` not working? Is there any solution so that I do not need to modify the user setting?

env info:
code-server: 3.5.0
python extension: 2021.10.1365161279 & 2020.8.101144 (Both versions have been tested and neither can work)

**Comment by jsjoeio** [comment]
Probably this! https://github.com/coder/code-server/issues/4221

**Reply by HumanZhong** [reply]
Exactly!
I found that my code-server version is actually 3.12.0 rather than the reported 3.5.0 and the situation seems very similar.

So, may I ask is there any suggested solution? Like should I change to 3.11.0 or this problem has been solved in 4.x so that i can upgrade to 4.x?

Thanks so much for your answer.

**Reply by jsjoeio** [reply]
Hmm...I'm not sure when it was working 🤔 3.11.0 could be worth trying. I haven't tested in 4.x, but you could test it out and see!

If you do try it out, will you let us know?

**Reply by HumanZhong** [reply]
I've tried 4.0.2 and it seems everything works fine so far. 
Thanks for your suggestion!

**Reply by jsjoeio** [reply]
Hooray!!! Thank you for reporting back :D Happy it's working!
